src/core/server/config/config_service.ts

```diff
@@ -139,6 +140,40 @@ export class ConfigService {
       );
     }
 
+    if (ConfigClass.deprecations !== undefined) {
```

Comment:
__note:__ initial version of config deprecations support for new platform configs. Since this PR is presumably targeting 7.0.0 we may just drop already deprecated properties in ES plugin removing the immediate need in this and rather handle deprecation support in the follow-up.

Reply:
Created PR https://github.com/elastic/kibana/pull/29496